orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pipeline/servergroup/strategies/RollingRedBlackStrategy.groovy

```diff
@@ -120,10 +118,29 @@ class RollingRedBlackStrategy implements Strategy, ApplicationContextAware {
       SyntheticStageOwner.STAGE_AFTER
     )
 
+    // Get source ASG from prior determineSourceServerGroupTask
     def source = null
+
     try {
-      source = getSource(targetServerGroupResolver, stageData, baseContext)
-    } catch (TargetServerGroup.NotFoundException e) {}
+      StageData.Source sourceAsg
+
+      Stage parentCreateAsgStage = stage.ancestors()
+                                        .find() { it.type == CreateServerGroupStage.PIPELINE_CONFIG_TYPE }
+
+      StageData parentStageData = parentCreateAsgStage.mapTo(StageData)
+      sourceAsg = parentStageData.source
+
+      if (sourceAsg != null && sourceAsg.serverGroupName != null) {
+        source = new ResizeStrategy.Source(
+          region: sourceAsg.region,
+          serverGroupName: sourceAsg.serverGroupName,
+          credentials: stageData.credentials ?: stageData.account,
+          cloudProvider: stageData.cloudProvider
+        )
+      }
+    } catch (Exception e) {
+      throw new IllegalStateException("Failed to locate source ASG while planning RRB flow", e)
```

Comment:
source server group while planning

Reply:
and we're consciously wanting to throw an exception? (previously it was just swallowed I guess?)

is there any context that you'd like to propagate in the exception message?

Reply:
i think an exception here is better - i'd rather it fail early than do something weird. Besides - this would truly be unexpected - nothing in here should throw an exception except for `parentCreateServerGroupStage` being null and then being deref'd which should not happen and if it does i'd rather it fail. we still allow an empty source server group (e.g. first deploy ever)

Reply:
Let me clean up the error message though. it can be made clearer